servicecontrol/troubleshooting.md

```diff
@@ -137,13 +137,21 @@ When building ServiceControl, all build artifacts are virus scanned to ensure no
 
 ## Stale indexes
 
-The database technology used for ServiceControl is based on asynchronous index updates. Indexes are not updated immediately, but very soon after data updates. A healthy system has indexes updated in milliseconds or up to several seconds under load. When indexes get very stale, this means that indexes lag behind for a long duration and can affect data presented and started tasks.
+Each ServiceControl service stores its data in a RavenDB embedded database. Indexes are used by RavenDB as way to query the documents. This indexing is done asynchronously in the background and is triggered whenever data is added or changed. The use of indexing is that it allows the server to respond quickly even when large amounts of data has changed and avoids costly table scan operations.
+
+But a downside of this is that as indexes are not updated immediately and can become stale quickly. A healthy system has indexes updated in milliseconds or up to several seconds under load. When indexes get very stale, this means that indexes lag behind for a long duration and can affect data presented and started tasks.
 
 Systems are affected by severe index lag when the following custom check message is presented:
 
 > At least one index significantly stale. Please run maintenance mode if this custom check persists to ensure index(es) can recover. See log file in `{LogPath}` for more details.
 
-After launching a ServiceControl instance in maintenance mode, message ingestion stops but the database engine still runs and messages will continue to queue. This ensures that any tasks related to index rebuilding or index scanning can run without interruption. This is useful when the storage isn't fast enough to do both message ingestion and index operations.
+A warning message is seen in the logs when the Indexing lag exceeds the default threshold value of 10,000
+
+```txt
+2023-07-03 09:41:08.4189|6|Warn|ServiceControl.CheckRavenDBIndexLag|Index [ExpiryKnownEndpointsIndex] IndexingLag 22,242 is above warning threshold (10,000). Launch in maintenance mode to let indexes catch up.
+```
+
+This can be usually resolved by launching  the ServiceControl instance in maintenance mode and letting the indexes catch up. Message ingestion stops but the database engine still runs and messages will continue to queue. This ensures that any tasks related to index rebuilding or index scanning can run without interruption. This is useful to resolve situations where the storage isn't fast enough to do both message ingestion and index operations, such as when an unexpected spike in message processing occurred.
```

Comment:
```suggestion
This can be resolved by launching the ServiceControl instance in maintenance mode and letting the indexes catch up. Message ingestion pauses when in maintenance mode, but the database engine still runs and messages will continue to queue. This ensures that any tasks related to index rebuilding or index scanning can run without interruption. This is useful to resolve situations where the storage isn't fast enough to do both message ingestion and index operations, such as when an unexpected spike in message processing occurred.
```